project/services/metadata.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2022 Indoc Research
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from enum import IntEnum
+
+import httpx
+from fastapi import Depends
+from requests import ConnectionError
+
+from project.components.exceptions import ServiceNotAvailable
+from project.components.exceptions import UnhandledException
+from project.config import Settings
+from project.config import get_settings
+
+logger = logging.getLogger(__name__)
+
+
+class Zones(IntEnum):
+    GREENROOM = 0
+    CORE = 1
+
+
+class MetadataClient:
+    """Client to connect with metadata service."""
+
+    def __init__(self, metadata_service_url):
+        self.service_url = metadata_service_url
+        self.zones = Zones
+
+    async def bulk_create_folder(self, users: list[dict], project_code: str) -> None:
```

Comment:
Please rename this method into a more concrete action and update also tests accordingly. 
```suggestion
    async def create_users_name_folders(self, users: list[dict[str, Any]], project_code: str) -> None:
```